aten/src/ATen/native/Activation.cpp

```diff
@@ -1,33 +1,242 @@
+#include <iostream>
 #include "ATen/ATen.h"
+#include "ATen/AccumulateType.h"
 #include "ATen/NativeFunctions.h"
+#include "ATen/TensorUtils.h"
+#include "ATen/WrapDimUtils.h"
+#include "ATen/native/cpu/ActivationKernel.h"
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#ifdef _MSC_VER
+#define LOG_SOFTMAX_SIZE_TYPE int64_t
```

Comment:
Why are we switching between signed and unsigned arithmetic? Can we just stick to signed arithmetic everywhere?

Reply:
Yeah I'll clean all this up. This is all legacy. I didn't port this.